### PR TITLE
Special case as_draws_array for length 1

### PR DESCRIPTION
**Pull request by rok-cesnovar**
When we use as_draws_array and we only have 1 chain we could optimize the function a bit. 

Testing scenario: 
```r
library(cmdstanr)
model_code <- "
parameters {
  real k;
}

transformed parameters {
  real x[50000] = rep_array(k, 50000);
}

model {
  k ~ normal(0, 1);
}
"
model_file <- write_stan_file(model_code)
mod <- cmdstan_model(model_file)
fit <- mod$sample(iter_sampling = 2000, validate_csv = FALSE, chains = 1)

data <- data.table::fread(cmd= paste0("grep -v '^#' ", fit$output_files()[1]))

times <- c()
for (i in 1:10) {
  start <- Sys.time()
  d_a <- posterior::as_draws_array(data)
  times <- c(times, as.double((Sys.time()-start), units = "secs"))
}
print(times)
```

this branch:
```
1.838615 1.740373 1.687281 1.830399 1.704403 1.841837 1.646028 1.733209 1.632276 1.731928
```
mean 1.74s

master:
```
2.565165 2.900125 2.978846 2.799916 2.859506 3.029445 2.660265 2.807306 2.583319 2.838322
```
mean 2.8s

So roughly 40% faster. 

This is important for cmdstanr where we read results one chain at a time. But probably makes sense anyway.
I am looking into other bottlenecks so if you prefer to review in larger chunks we can keep the PR open and I can note when I am done.

### CODE REVIEW

File: R/as_draws_array.R
Comment (rok-cesnovar) [inline R/as_draws_array.R:64]: The biggest bottleneck now is the as.matrix call below, but that probably cant be avoided.

Comment (paul-buerkner) [reply R/as_draws_array.R:64]: Indeed. We have to transform a list-like object (columns are elements of a list) into a numeric matrix in column major order. I am not sure we can do this faster than base R here.

**Comment by paul-buerkner** [comment]
Thank you! I am fine with both, keeping the PR open for more changes or merging just this fix. What is your preference?

File: R/as_draws_array.R
Comment (rok-cesnovar) [reply R/as_draws_array.R:64]: Yeah, I figured.

**Comment by rok-cesnovar** [comment]
I think this one is worth it on its own. Will open subsequent ones if something worthy comes up.

**Review by paul-buerkner — APPROVED** [review]
The PR looks good to me. Thank you! @jgabry do you want to take a look before we merge?

**Comment by jgabry** [comment]
Thanks! It looks good to me too. Should we bump the version number when we merge this? That way we can require this version for cmdstanr.

**Comment by rok-cesnovar** [comment]
Lets hold off for a few more days with the bump, I might have time to look into other bottlnecks causing #98 
If I dont find any till next week, we can then bump.

**Comment by jgabry** [comment]
Ok, sounds good thanks!

**Comment by jgabry** [comment]
Since @paul-buerkner already approve this I'll go ahead and merge now and then we can do separate PRs for other fixes (e.g., if you or someone else figures out #98)

**Pull request merged**
